Month3/PixelGameEngine/AsteroidGame/AsteroidMain.cpp

```diff
@@ -0,0 +1,463 @@
+	}
+
+	bool WinGame()
+	{
+		int MaxLevel = 4;
+		if (Level >= MaxLevel)
+		{
+			ModelShip.clear();
+			ModelAsteroids.clear();
+			Bullets.clear();
+
+			DrawString(60, 100, "You Win", olc::GREEN, 10);
+			return true;
+		}
+		else
+		{
+			return false;
+		}
+
+	}
+	void LevelUp()
+	{
+		if (VecAsteroid.empty())
+		{
+			Level += 1;
+			// Level Clear
+			Score += 1000; // Large score for level progression
+			VecAsteroid.clear();
+			Bullets.clear();
+			if (Level == 2)
+			{
+				VecAsteroid.push_back({ 30.0f * sinf(Player.Angle - 3.14159f / 2.0f) + Player.PosX,
+												  30.0f * cosf(Player.Angle - 3.14159f / 2.0f) + Player.PosY,
+												  10.0f * sinf(Player.Angle), 10.0f * cosf(Player.Angle),(int)15, 0.0f,false });
+
+
+				VecAsteroid.push_back({ 30.0f * sinf(Player.Angle - 3.14159f / 2.0f) + Player.PosX,
+												  40.0f * cosf(Player.Angle - 3.14159f / 2.0f) + Player.PosY,
+												  40.0f * sinf(Player.Angle), 10.0f * cosf(Player.Angle),(int)15, 0.0f,false });
+			}
+			else if (Level == 3)
+			{
+				VecAsteroid.push_back({ 30.0f * sinf(Player.Angle - 3.14159f / 2.0f) + Player.PosX,
+												  40.0f * cosf(Player.Angle - 3.14159f / 2.0f) + Player.PosY,
+												  10.0f * sinf(Player.Angle), 10.0f * cosf(Player.Angle),(int)15, 0.0f,false });
+
+
+				VecAsteroid.push_back({ 30.0f * sinf(Player.Angle - 3.14159f / 2.0f) + Player.PosX,
+												  40.0f * cosf(Player.Angle - 3.14159f / 2.0f) + Player.PosY,
+												  10.0f * sinf(Player.Angle), 10.0f * cosf(Player.Angle),(int)15, 0.0f,false });
+				VecAsteroid.push_back({ 100.0f * sinf(Player.Angle - 3.14159f / 2.0f) + Player.PosX,
+												  40.0f * cosf(Player.Angle - 3.14159f / 2.0f) + Player.PosY,
+												  10.0f * sinf(Player.Angle), 10.0f * cosf(Player.Angle),(int)15, 0.0f,false });
+
+			}
+			else if (Level == 4)
+			{
+				VecAsteroid.push_back({ 30.0f * sinf(Player.Angle - 3.14159f / 2.0f) + Player.PosX,
+												  30.0f * cosf(Player.Angle - 3.14159f / 2.0f) + Player.PosY,
+												  10.0f * sinf(Player.Angle), 10.0f * cosf(Player.Angle),(int)15, 0.0f,false });
+
+
+				VecAsteroid.push_back({ 30.0f * sinf(Player.Angle - 3.14159f / 2.0f) + Player.PosX,
+												  30.0f * cosf(Player.Angle - 3.14159f / 2.0f) + Player.PosY,
+												  10.0f * sinf(Player.Angle), 10.0f * cosf(Player.Angle),(int)15, 0.0f,false });
+
+				VecAsteroid.push_back({ 100.0f * sinf(Player.Angle - 3.14159f / 2.0f) + Player.PosX,
+												  40.0f * cosf(Player.Angle - 3.14159f / 2.0f) + Player.PosY,
+												  10.0f * sinf(Player.Angle), 10.0f * cosf(Player.Angle),(int)15, 0.0f,false });
+
+				VecAsteroid.push_back({ 70.0f * sinf(Player.Angle - 3.14159f / 2.0f) + Player.PosX,
+												  40.0f * cosf(Player.Angle - 3.14159f / 2.0f) + Player.PosY,
+												  10.0f * sinf(Player.Angle), 10.0f * cosf(Player.Angle),(int)15, 0.0f,false });
+
+
+			}
+
+		}
+	}
```

Comment:
you could create each level in a function, furthermore with a random() and a for you could create the different amounts of asteroids and positions